mercure.rst

```diff
@@ -0,0 +1,564 @@
+updates to the Hub.
+
+The `Publisher` service can be injected using the
+:doc:`autowiring </service_container/autowiring>` in any other
+service, including controllers:
+
+.. code-block:: php
+
+    // src/Controller/PublishController.php
+    namespace App\Controller;
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Mercure\Publisher;
+    use Symfony\Component\Mercure\Update;
+
+    class PublishController
+    {
+        public function __invoke(Publisher $publisher): Response
+        {
+            $update = new Update(
+                'http://example.com/books/1',
+                json_encode(['status' => 'OutOfStock'])
+            );
+
+            // The Publisher service is an invokable object
+            $publisher($update);
+
+            return new Response('published!');
+        }
+    }
+
+The first parameter to pass to the `Update` constructor is
+the **topic** being updated. This topic should be an IRI_
+(Internationalized Resource Identifier, RFC 3987): a unique identifier
+of the resource being dispatched.
+
+Usually, this parameter contains the original URL of the resource
+transmitted to the client, but it can be any valid IRI, it doesn't
+have to be an URL that exists (similarly to XML namespaces).
+
+The second parameter of the constructor is the content of the update.
+It can be anything, stored in any format.
+However, serializing the resource in a hypermedia formats such as JSON-LD,
+Atom, HTML or XML is recommended.
+
+Subscribing
+~~~~~~~~~~~
+
+Subscribing to updates in JavaScript is straightforward:
+
+.. code-block:: html
+
+    <script>
+    const es = new EventSource('http://localhost:3000/hub?topic=' + encodeURIComponent('http://example.com/books/1'));
+    es.onmessage = e => {
+        // Will be called every time an update is published by the server
+        console.log(JSON.parse(e.data));
+    }
+    </script>
+
+Mercure also allows to subscribe to several topics,
+and to use URI Templates as patterns:
+
+.. code-block:: html
+
+    <script>
+    const u = new URL('http://localhost:3000/hub'); // URL is a built-in JavaScript class to manipulate URLs
+    u.searchParams.append('topic', 'http://example.com/books/1');
+    u.searchParams.append('topic', 'http://example.com/books/2'); // Subscribe to updates of several Book resources
+    u.searchParams.append('topic', 'http://example.com/reviews/{id}'); // All Review resources will match this pattern
+
+    const es = new EventSource(u);
+    es.onmessage = e => {
+        console.log(JSON.parse(e.data));
+    }
+    </script>
+
+.. tip::
+
+    Google Chrome DevTools natively integrate a `practical UI`_ displaying in live
+    the received events:
+
+    .. image:: /_images/mercure/chrome.png
+
+    To use it:
+    * open the DevTools
+    * select the "Network" tab
+    * click on the request to the Mercure hub
+    * click on the "EventStream" sub-tab.
+
```

Comment:
Maybe add a mention of https://github.com/dunglas/uri-template-tester to test the templates?